starknet/src/display.rs

```diff
@@ -0,0 +1,368 @@
+use ledger_device_sdk::ui::{
+    bitmaps::{Glyph, BACK, CERTIFICATE, CROSSMARK, DASHBOARD_X, EYE, VALIDATE_14, WARNING},
+    gadgets::{
+        clear_screen, EventOrPageIndex, Field, MultiFieldReview, MultiPageMenu, Page, PageStyle,
+    },
+};
+
+#[cfg(any(target_os = "stax", target_os = "flex"))]
+use ledger_device_sdk::nbgl::{
+    Field, NbglGenericReview, NbglGlyph, NbglHomeAndSettings, NbglPageContent, NbglReview,
+    NbglReviewStatus, NbglStatus, TagValueConfirm, TagValueList, TuneIndex,
+};
+
+use crate::Ins;
+
+pub fn show_tx(ctx: &mut Ctx) -> Option<bool> {
+    match support_clear_sign(&ctx.tx) {
+        Some(t) => {
+            testing::debug_print("Clear sign supported !!! \n");
+            let tx = &ctx.tx;
+            let call = &tx.calls[0];
+
+            let sender = tx.sender_address.to_hex_string();
+            let token = ERC20_TOKENS[t].ticker;
+            let to = call.calldata[0].to_hex_string();
+            let amount = call.calldata[1].to_dec_string(Some(ERC20_TOKENS[t].decimals));
+
+            testing::debug_print("Compute fees \n");
+
+            let max_amount = FieldElement::from(&tx.l1_gas_bounds.value[8..16]);
+            let max_price_per_unit = FieldElement::from(&tx.l1_gas_bounds.value[16..32]);
+            let max_fees = max_amount * max_price_per_unit;
+            let mut max_fees_str = max_fees.to_dec_string(None);
+            max_fees_str.push_str(" fri");
+
+            testing::debug_print("Compute fees OK \n");
+
+            let my_fields = [
+                Field {
+                    name: "From",
+                    value: sender.as_str(),
+                },
+                Field {
+                    name: "Token",
+                    value: token,
+                },
+                Field {
+                    name: "Amount",
+                    value: amount.as_str(),
+                },
+                Field {
+                    name: "To",
+                    value: to.as_str(),
+                },
+                Field {
+                    name: "Max Fees",
+                    value: max_fees_str.as_str(),
+                },
+            ];
+
+            testing::debug_print(&format!(
+                "Token: {}\nTo: {}\nAmount: {}\n",
+                token, to, amount
+            ));
+
+            #[cfg(not(any(target_os = "stax", target_os = "flex")))]
+            {
+                let my_review = MultiFieldReview::new(
+                    &my_fields,
+                    &["Confirm Tx to sign"],
+                    Some(&EYE),
+                    "Approve",
+                    Some(&VALIDATE_14),
+                    "Reject",
+                    Some(&CROSSMARK),
+                );
+
+                /*match my_review.show() {
+                    true => {
+                        ctx.hash.m_hash = crypto::tx_hash(&ctx.tx);
+                        true
+                    }
+                    false => false,
+                }*/
```

Comment:
Can this code (and the ones below) be removed if not needed anymore? Seems handled by the caller.